xbmc/Util.cpp

```diff
@@ -1834,18 +1820,41 @@ void CUtil::ScanForExternalSubtitles(const std::string& strMovie, std::vector<st
     || item.IsLiveTV()
     || !item.IsVideo())
     return;
-  
+
+  CLog::Log(LOGDEBUG,"%s: Searching for subtitles...", __FUNCTION__);
+
+  std::string strBasePath;
+  std::string strSubtitle = URIUtils::ReplaceExtension(URIUtils::GetFileName(strMovie), "");
+
+  if (item.HasVideoInfoTag())
+    strBasePath = item.GetVideoInfoTag()->m_basePath;
+
+  if (item.IsOpticalMediaFile())
+    strBasePath = item.GetLocalMetadataPath();
+
+  if (strBasePath.empty())
+    strBasePath = URIUtils::GetBasePath(strMovie);
+
+  CFileItemList items;
   vector<std::string> strLookInPaths;
-  
-  std::string strMovieFileName;
-  std::string strMoviePath;
-  
-  URIUtils::Split(strMovie, strMoviePath, strMovieFileName);
-  std::string strMovieFileNameNoExt(URIUtils::ReplaceExtension(strMovieFileName, ""));
-  strLookInPaths.push_back(strMoviePath);
-  
+
+  int flags = DIR_FLAG_NO_FILE_DIRS | DIR_FLAG_READ_CACHE | DIR_FLAG_NO_FILE_INFO;
```

Comment:
Since you use this as a const, better declare const. Personally I'd drop the var all together and just specify the flags directly in the 2 functions below.